frontend/public/extend/devconsole/components/topology/shapes/PodStatus.tsx

```diff
@@ -17,92 +18,105 @@ export const podColor = {
+  'Unknown',
+];
+
 type PodStatusProps = {
-  x?: number;
-  y?: number;
-  pods: Pod[];
   innerRadius: number;
   outerRadius: number;
+  size: number;
+  standalone: boolean;
+  x?: number;
+  y?: number;
+  data: Pod[];
+  showTooltip?: boolean;
 };
 
-export default class PodStatus extends React.PureComponent<PodStatusProps> {
-  willLeave = ({ style }) => {
-    return {
-      ...style,
-      startAngle: style.endAngle,
-    };
-  };
-
-  willEnter = ({ style }) => {
-    return {
-      ...style,
-      endAngle: style.startAngle,
-    };
-  };
-
-  chooseColor = (d) => {
-    return { fill: podColor[d.status.phase] };
-  };
-
-  render() {
-    const { pods, innerRadius, outerRadius, x = 0, y = 0 } = this.props;
-    const pieFunc = pie().sort(null);
-
-    const podData = pods.map(() => 100 / pods.length);
-
-    const arcFunc = arc()
-      .outerRadius(outerRadius)
-      .innerRadius(innerRadius)
-      .padAngle(0.02);
-
-    const pieData = pieFunc(podData);
-
-    const motionStyles = pieData.map((d, i) => ({
-      key: `${i}`,
-      data: { ...d, index: i, podData: this.props.pods[i] },
-      style: d,
-    }));
-
-    const defaultStyles = pieData.map((d, i) => ({
-      key: `${i}`,
-      data: { ...d, index: i, podData: this.props.pods[i] },
-      style: { ...d, endAngle: d.startAngle },
-    }));
-
-    const centerTransform = `translate(${x}, ${y})`;
+const PodStatus: React.FunctionComponent<PodStatusProps> = ({
+  innerRadius,
+  outerRadius,
+  x,
+  y,
+  data,
+  size,
+  standalone,
+  showTooltip = true,
+}) => {
+  const vData = [];
+  podStatus.forEach((pod) => {
+    let podNumber = 0;
+    data.forEach((element) => {
+      if (element.status.phase === pod) {
+        podNumber += 1;
+      }
+    });
+    if (podNumber !== 0) {
+      vData.push({ x: pod, y: podNumber });
+    }
+  });
+  const centerTransform = `translate(${x}, ${y})`;
+  return (
+    <VictoryPie
+      events={
+        showTooltip
+          ? [
+            {
+              target: 'data',
+              eventHandlers: {
+                onMouseOver: () => {
+                  return [
+                    {
+                      target: 'labels',
+                      mutation: (props) => {
+                        return { active: true };
+                      },
+                    },
+                  ];
+                },
+                onMouseOut: () => {
+                  return [
+                    {
+                      target: 'labels',
+                      mutation: (props) => {
+                        return { active: false };
+                      },
+                    },
+                  ];
+                },
+              },
+            },
+          ]
+          : []
+      }
+      animate={{
+        duration: 2000,
+      }}
+      standalone={standalone}
+      innerRadius={innerRadius}
+      radius={outerRadius}
+      groupComponent={x && y ? <g transform={centerTransform} /> : <g />}
```

Comment:
```suggestion
      groupComponent={x && y ? <g transform={`translate(${x}, ${y})`} /> : <g />}
```
Do you have to provide the else clause as `<g />` or will `undefined` or `null` work?

Reply:
null doesn't but undefined does. fixed https://github.com/talamer/console/pull/98/commits/116ad0e7e32c3724225c931a7528ea4fdf0ff6e1